### ISSUE TITLE
Resource name with uppercase letters leads to wrong class being resolved

### ISSUE THREAD

**Issue by cypressious** [lagoshny/ngx-hateoas-client#45]
I have a resource and a service

```ts
@HateoasResource('fooBar')
export class FooBar extends Resource {
  date: string;

  formatDate(format: string) {
    return dayjs(this.date, ISO_DATE_FORMAT).format(format);
  }
}

@Injectable({ providedIn: 'root' })
export class FooBarService extends RestService<FooBar> {
  constructor() {
    super(FooBar);
  }
}
```

When I call `fooBarService.getPage()`, the returned collections contains objects of type `Resource`, not `FooBar` which leads to errors when I try to call `formatDate()` on them.

I have stepped through the code and it seems that this is caused by the uppercase letter in the resource name.

**Comment by cypressious** [comment]
This seems to be caused by the fix for #26 in #27.

**Comment by cypressious** [comment]
Thank you, I can confirm the bug is fixed.
